src/Codeception/Module/Yii2.php

```diff
@@ -276,7 +277,9 @@ public function _initialize(): void
 
         $this->defineConstants();
         $this->server = $_SERVER;
-        $this->initServerGlobal();
+        // Adds the required server params. Note this is done separately from the request cycle since someone might call
+        // `Url::to` before doing a request, which would instantiate the request component with incorrect server params.
+        $_SERVER = array_merge($_SERVER, $this->getServerParams());
```

Comment:
Use array destructuring
```suggestion
        $_SERVER = [...$_SERVER, $this->getServerParams()];
```

Reply:
Done ✅.